### PR TITLE
feat: Dynamic Dashboard Sections 

### PR DESCRIPTION
**Pull request by efstajas**
Makes the dashboard fetch workstreams in a number of possible states and dynamically displays the results in "sections", depending on whether or not the section's respective query has returned any workstreams.

In practice, this should lead to the dashboard automatically showing all of the workstreams relevant to the user.

For example, after applying to one workstream and creating another, which no-one has yet applied to, the dashboard looks like this:

Screenshot 2022-05-02 at 14 43 23

After someone applies to the user's workstream, it looks like this:
Screenshot 2022-05-02 at 14 45 23

Note how the "Applications to review" section now appears above the "Workstreams you applied for" section, as it is deemed more important.

### CODE REVIEW

File: src/lib/components/dashboard/Section.svelte [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    </div>
+    {#if earningPerDay}
+      <div class="earning-per-day">
+        <TokenStreams />
+        <p>
+          You are earning <span class="typo-text-bold">{earningPerDay} DAI</span
+          > / day
+        </p>
+      </div>
+    {/if}
```
Comment (juliendonck) [inline src/lib/components/dashboard/Section.svelte:25]: Shouldn't this just be a `<Slot />` Seems more flexible/general purpose to me.

Comment (efstajas) [reply src/lib/components/dashboard/Section.svelte:25]: 👍👍 a named slot would make sense here. Will adjust that

Comment (juliendonck) [reply src/lib/components/dashboard/Section.svelte:25]: Nice, also is there a way that prettier converts all `px` to `rem` on save or something?

Comment (efstajas) [reply src/lib/components/dashboard/Section.svelte:25]: That'd be great, but couldn't find any with a short Google :D I suppose I just need to get used to that, just keep pointing it out ;)

**Review by juliendonck — APPROVED** [review]
LGTM

**Pull request merged**